libsplinter/src/auth/oauth/mod.rs

```diff
@@ -93,6 +93,43 @@ impl OAuthClient {
         })
     }
 
+    /// Creates a new `OAuthClient` with GitHub's authorization and token URLs.
+    ///
+    /// # Arguments
+    ///
+    /// * `client_id` - The GitHub OAuth client ID
+    /// * `client_secret` - The GitHub OAuth client secret
+    /// * `redirect_url` - The endpoint that GitHub will redirect to after it has completed
+    ///   authorization
+    #[cfg(feature = "oauth-github")]
+    pub fn github(
+        client_id: String,
+        client_secret: String,
+        redirect_url: String,
+    ) -> Result<Self, OAuthClientConfigurationError> {
+        let client =
+            BasicClient::new(
+                ClientId::new(client_id),
+                Some(ClientSecret::new(client_secret)),
+                AuthUrl::new("https://github.com/login/oauth/authorize".into())
+                    .expect("Authorization URL should be valid"),
```

Comment:
Replace these expects with OAuthClientConfigurationError::InvalidXUrl() like in new(). Or maybe just have this function call new()?

Reply:
More forgotten expects. Thanks again. Updated to just call `Self::new`